adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepository.java

```diff
@@ -482,4 +486,43 @@ public FareRule getFareRule(final String fareId, final String routeId, final Str
         return fareRuleCollection.get(FareRule.getFareRuleMappingKey(fareId, routeId, originId, destinationId,
                 containsId));
     }
+
+    /**
+     * Add a shape point representing a row from shapes.txt to this. Return the entity added to the repository if not
+     * null, otherwise returns null.
+     *
+     * @param newShapePoint the internal representation of a row from shapes.txt to be added to the repository.
+     * @return the entity added to the repository if not null, otherwise returns null.
+     * @throws IllegalArgumentException if the shape point passed as argument is null
+     */
+    @Override
+    public ShapePoint addShapePoint(final ShapePoint newShapePoint) throws IllegalArgumentException {
```

Comment:
Here again, maybe the distinction between what's a Shape and what is a ShapePoint could be made clearer